src/overlays/actors/ovl_Obj_Lightblock/z_obj_lightblock.c

```diff
@@ -22,46 +27,178 @@ const ActorInit Obj_Lightblock_InitVars = {
-static ColliderCylinderInit D_80AF3EA0 = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER | AC_TYPE_OTHER, OC1_NONE, OC2_NONE, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x00202000, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER | AC_TYPE_OTHER,
+        OC1_NONE,
+        OC2_NONE,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00202000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_NONE,
+    },
     { 84, 120, 0, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80AF3EEC[] = {
+typedef struct {
+    /* 0x00 */ f32 scale;
+    /* 0x04 */ s16 radius;
+    /* 0x06 */ s16 height;
+    /* 0x08 */ s16 yShift;
+    /* 0x0C */ s32 params;
+} LightblockTypeVars; // size = 0x10
+
+static LightblockTypeVars sLightblockTypeVars[] = {
+    { 0.1f, 76, 80, 19, 2 },
+    { (1.0f / 6), 126, 144, 19, 3 },
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 500, ICHAIN_STOP),
 };
 
-#endif
-
-extern ColliderCylinderInit D_80AF3EA0;
-extern InitChainEntry D_80AF3EEC[];
-
-extern UNK_TYPE D_06000178;
-extern UNK_TYPE D_06000B80;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3910.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/ObjLightblock_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/ObjLightblock_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3AC8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3ADC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3B8C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3BA0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3C18.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3C34.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/ObjLightblock_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/ObjLightblock_Draw.s")
+extern ColliderCylinderInit sCylinderInit;
+extern InitChainEntry sInitChain[];
+
+extern CollisionHeader D_06000B80;
+extern Gfx D_801AEF88[];
+extern Gfx D_06000178[];
+extern Gfx D_801AEFA0[];
+
+void func_80AF3910(ObjLightblock* this, GlobalContext* globalCtx) {
+    LightblockTypeVars* typeVars = &sLightblockTypeVars[LIGHTBLOCK_TYPE(&this->dyna.actor)];
+
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, 0x48, this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y,
```

Comment:
Try to use the enum values defined in `include/z64actor.h`
```suggestion
    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_DEMO_EFFECT, this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y,
```